### DISCUSSION TITLE
Popping / clicking sound when creating audio driver on Windows 10

### DISCUSSION THREAD

**Discussion by JBetz** [FluidSynth/fluidsynth#1246] [Q&A] [answered]
On Windows 10, I always hear a popping / clicking sound for a split second when calling `fluid_synth_new_audio_driver`. It only happens once, and only during initialization, but it's rather jarring and I want to get rid of it. Switching the driver from `dsound` to `wasapi` results in a slightly softer clicking sound, but it's still noticeable.

How do I go about debugging this?

**Comment by derselbst** [comment, marked as answer]
`new_fluid_audio_driver` (!) messes with the underlying audio APIs, hence I'm not really surprised that it results in some audible glitch. I would assume that this function is only called once in the entire lifetime of a program, so I don't quite comprehend why you want to debug it. Anyway, you should first try whether it happens on a different computer as well (or with a different soundcard). If it does, you should provide a minimal reproducible code example to see if others have the same problem.

I can't think of a good way of debugging this. Rather only by putting log statements into fluidsynths code and commenting out functions / codeblocks to nail down the function where this is happening.
